pkg/console/operator/operator.go

```diff
@@ -135,84 +129,55 @@ func (c *consoleOperator) Key() (metav1.Object, error) {
 	return operatorConfig, err
 }
 
-// sync() is the handler() function equivalent from the sdk
-// this is the big switch statement.
-// TODO: clean this up a bit, its messy
 func (c *consoleOperator) Sync(obj metav1.Object) error {
 	startTime := time.Now()
 	logrus.Infof("started syncing operator %q (%v)", obj.GetName(), startTime)
 	defer logrus.Infof("finished syncing operator %q (%v) \n\n", obj.GetName(), time.Since(startTime))
 
-	operatorConfig := obj.(*consolev1alpha1.Console)
+	operatorConfig := obj.(*consolev1.Console)
 
-	switch operatorConfig.Spec.ManagementState {
-	case operatorsv1alpha1.Managed:
+	if err := c.handleSync(operatorConfig); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *consoleOperator) handleSync(config *consolev1.Console) error {
+
+	switch config.Spec.ManagementState {
+	case operatorv1.Managed:
 		logrus.Println("console is in a managed state.")
 		// handled below
-	case operatorsv1alpha1.Unmanaged:
+	case operatorv1.Unmanaged:
 		logrus.Println("console is in an unmanaged state.")
 		return nil
-	// take a look @ https://github.com/openshift/service-serving-cert-signer/blob/master/pkg/operator/operator.go#L86
-	case operatorsv1alpha1.Removed:
+		// take a look @ https://github.com/openshift/service-serving-cert-signer/blob/master/pkg/operator/operator.go#L86
+	case operatorv1.Removed:
 		logrus.Println("console has been removed.")
-		return c.deleteAllResources(operatorConfig)
+		return c.deleteAllResources(config)
 	default:
 		// TODO should update status
-		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+		return fmt.Errorf("unknown state: %v", config.Spec.ManagementState)
 	}
 
-	var currentActualVersion *semver.Version
-
-	// TODO: ca.yaml needs a version, update the v1alpha1.Console to include version field
-	if ca := operatorConfig.Status.CurrentAvailability; ca != nil {
```

Comment:
Most of this was a guess at what would be useful for deciding version.  We can deal with version when its an issue & its clearer what the requirements will be.